scalding-core/src/test/scala/com/twitter/scalding/source/typedtext/TypedTextTest.scala

```diff
@@ -0,0 +1,24 @@
+package com.twitter.scalding.source.typedtext
+
+import com.twitter.scalding._
+
+import org.scalatest.FunSuite
+import org.scalatest.junit.JUnitRunner
+
+case class Test1(a: Int, b: Long, c: Option[Double])
+case class Test2(one: Test1, d: String)
+
+class TypedTextTest extends FunSuite {
+  test("Test with a flat tuple") {
+    val source = TypedText.tsv[Test1]("myPath")
+    assert(source.sourceFields.size == 3)
+  }
+  test("Test with a nested tuple") {
+    val source = TypedText.tsv[Test2]("myPath")
```

Comment:
It would be good to just write out with those types (Int, Int), and String. Manually verify to yourself once. But then in the test you can read them back into matching case classes to see things round trip as you expect. Or for String cases use the TextLine source and it should be the same then too